gen_gadgets.sh

```diff
@@ -0,0 +1,99 @@
+cat <<EOF > gadgets.h
```

Comment:
I wonder if this should be a c file instead? I'm not a big fan including giant data structures in headers because they are pulled in every time the header is included.

Probably generating `gadgets.c` with the content and `gadgets.h` with the reference would be a good idea.